workflows/prognostic_c48_run/runtime/overrider.py

```diff
@@ -0,0 +1,130 @@
+import dataclasses
+import functools
+import logging
+from typing import Hashable, Mapping, MutableMapping, Set
+
+import cftime
+import fsspec
+import xarray as xr
+
+import fv3gfs.util
+from runtime.monitor import Monitor
+from runtime.types import Diagnostics, Step
+from runtime.derived_state import DerivedFV3State
+from runtime.utils import quantity_state_to_ds, ds_to_quantity_state
+
+QuantityState = MutableMapping[Hashable, fv3gfs.util.Quantity]
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class OverriderConfig:
+    """Configuration for overriding tendencies from a step.
+    
+    Attributes:
+        url: path to dataset containing tendencies.
+        variables: mapping from state name to name of corresponding tendency in
+            provided dataset. For example: {"air_temperature": "fine_res_Q1"}.
+    """
+
+    url: str
+    variables: Mapping[str, str]
+
+
+@dataclasses.dataclass
+class OverriderAdapter:
+    """Wrap a Step function to override certain tendencies."""
+
+    config: OverriderConfig
+    state: DerivedFV3State
+    communicator: fv3gfs.util.CubedSphereCommunicator
+    timestep: float
+    diagnostic_variables: Set[str] = dataclasses.field(default_factory=set)
+
+    def __post_init__(self: "OverriderAdapter"):
+        if self.communicator.rank == 0:
+            logger.debug(f"Opening tendency overriding dataset from: {self.config.url}")
+        tile = self.communicator.partitioner.tile_index(self.communicator.rank)
+        mapper = fsspec.get_mapper(self.config.url)
+        ds = xr.open_zarr(mapper, consolidated=True)
+        ds = ds[list(self.config.variables.values())].isel(tile=tile)
+        self._tendency_ds = DatasetCachedByChunk(ds, "time")
+
+    def _open_tendencies_dataset(self, time: cftime.DatetimeJulian) -> xr.Dataset:
+        ds = xr.Dataset()
+        if self.communicator.tile.rank == 0:
+            ds = self._tendency_ds.load(time).drop_vars("time")
+        tendencies = self.communicator.tile.scatter_state(ds_to_quantity_state(ds))
+        return quantity_state_to_ds(tendencies)
+
+    @property
+    def monitor(self) -> Monitor:
+        return Monitor.from_variables(
+            self.diagnostic_variables, self.state, self.timestep
+        )
+
+    def override(self, name: str, func: Step) -> Diagnostics:
```

Comment:
Not used publicly so I would opt to hide it.

```suggestion
    def _override(self, name: str, func: Step) -> Diagnostics:
```